src/shared/utils/vis.ts

```diff
@@ -376,3 +377,37 @@ export const getMainColumnName = (
   }
   return ''
 }
+
+const getS2CellID = (table, index: number): string => {
+  const column = table.getColumn('s2_cell_id')
+  if (!column) {
+    return null
+  }
+  const value = column[index]
+  if (typeof value !== 'string') {
+    return null
+  }
+  return value
+}
+
+export const PRECISION_TRIMMING_TABLE = [BigInt(1)]
+for (let i = 1; i < 17; i++) {
+  PRECISION_TRIMMING_TABLE[i] = PRECISION_TRIMMING_TABLE[i - 1] * BigInt(16)
+}
```

Comment:
What's up with this `for` loop? what is `17`? should `PRECISION_TRIMMING_TABLE` be initialized with the proper values when it's exported?